tests/test_requests.py

```diff
@@ -32,10 +33,29 @@
 from .compat import StringIO, u
 from .utils import override_environ
 
+from requests.packages.urllib3.exceptions import SNIMissingWarning
+
+# urllib3 sets this to only go off once, but we need it to
+# always fire for test_https_warnings to work
+warnings.simplefilter('always', SNIMissingWarning)
```

Comment:
Maybe set this in a context manager around the single test that needs this. E.g.,

```py
@contextlib.contextmanager
def enable_sni_missing():
    warnings.simplefilter('always', SNIMissingWarning)
    yield
    warnings.simplefilter('once', SNIMissingWarning)
```

(Pseudo-code warning)